src/main/scala/com/codacy/client/bitbucket/client/BitbucketClient.scala

```diff
@@ -32,9 +32,39 @@ class BitbucketClient(credentials: Credentials) {
     }
   }
 
-  /*
-   * Does a paginated API request and parses the json output into a sequence of classes
-   */
+  /**
+    * Make an API request and parse the json output into a response. This response includes not only the objects
+    * requested but pagination information as well.
+    *
+    * @param request The information of the request to be performed
+    * @param reader A [[Reads]] to parse the json response from the API
+    * @tparam T The type of the objects in the response (excluding pagination information)
+    * @return A [[SuccessfulResponse]] or a [[FailedResponse]] depending if the request was successful or not
+    */
+  def executeWithCursor[T](request: Request[T])(implicit reader: Reads[T]): RequestResponse[Seq[T]] = {
```

Comment:
We should have request: Request[Seq[T]] as the executePaginated to garantee that we only call this for endpoints that return multiple items